### PR TITLE
Genbank has removed GI identifier in the genbank files

### PR DESCRIPTION
**Pull request by thibauthourlier**
Do not look for GI in VERSION in the first regex. Still looking for GI if line has VERSION
Bug fix: Call get_accession instead of direct hash access to avoid accession to not be initialised

### CODE REVIEW

**Comment by thibauthourlier** [comment]
I have not updated modules/t/input/data.gbk because I thought it might be better to fetch the file with the accession so we can know when the format changes. I know it might not be the best option...

File: modules/Bio/EnsEMBL/IO/Parser/Genbank.pm [hunk 1]
```diff
@@ -92,13 +92,16 @@ sub read_record {
             $self->{'record'}->{'_raw_accession'} = $field;
         }
         elsif ($field_type eq 'VERSION') {
-            if ($field =~ /\S+\.(\d+)\s+GI:(\d+)/) {
+            if ($field =~ /\S+\.(\d+)/) {
                 $self->{'record'}->{'_version'} = $1;
                 $self->{'record'}->{'_genebank_id'} = $2;
```
Comment (premanand17) [inline modules/Bio/EnsEMBL/IO/Parser/Genbank.pm]: There is no capture for $2 in ($field =~ /\S+\.(\d+)/).  Think _genebank_id will never be initialised there?!

Comment (thibauthourlier) [reply modules/Bio/EnsEMBL/IO/Parser/Genbank.pm]: You are correct this line should have been removed

**Comment by nerdstrike** [comment]
Also, fetching sample files from remote resources for tests causes way more trouble than its worth. We've suffered from this behaviour elsewhere. The purpose of the canned data is to prevent code regressions from us, not validate remote changes in format.

**Comment by magaliruffier** [comment]
Let's pretend there is a PR template for this repository, can you please give us a bit more background?
What is the change, why is it needed, where is it needed, what are the consequences of not applying the change, what are the consequences of applying the change and how did you test for it?

**Comment by thibauthourlier** [comment]
**What is the change**
Only have one capture block in the regex which look for the version of the file. Try to get the Genbank identifier if GI is found on the line containing VERSION
**Why is it needed**
Genbank has removed the Genbank identifier from their databases and their files
**Where is it needed**
In the read_record method of the Genbank parser
**What are the consequences of not applying the change**
The version number of the accession cannot be retrieved
**What are the consequences of applying the change**
The version number of the accession can be retrieved
**How did you test for it**
By parsing a Genbank file to load the mitochondria for catfish

**What is the change**
Replacement of a direct access to $self->{'record'}->{'_accession'} with $self->get_accession
**Why is it needed**
If $self->get_sequence_name is call before $self->get_accession, get_sequence_name only return the version instead of "accession.version"
**Where is it needed**
In the get_sequence_name method of the Genbank parser
**What are the consequences of not applying the change**
The versionned accession cannot be retrieved
**What are the consequences of applying the change**
The versionned accession can be retrieved
**How did you test for it**
By parsing a Genbank file to load the mitochondria for catfish

**Comment by thibauthourlier** [comment]
Hi,
I've added some tests, updated the test data file and added few more little changes

**Comment by avullo** [comment]
That all sounds reasonable enough, thanks Thibaut.

**Pull request merged**